hw8/script.js

```diff
@@ -0,0 +1,190 @@
+const newArrayOfNumbers = [];
+const arrayLeтgth = Math.abs(
+  +prompt("Ведіть довжину масиву який ви хочете створити")
+);
+if (isNaN(arrayLeтgth)) {
+  console.log("Не вірний тип даних!");
+} else {
+  newArrayOfNumbers.length = arrayLeтgth;
+}
+for (let i = 0; i < newArrayOfNumbers.length; i++) {
+  const userNumber = +prompt(
+    `Ведіть число яке ви хочете додати до масиву під індексом ${i}`
+  );
+  if (isNaN(userNumber)) {
+    console.log("Не вірний тип даних!");
+    newArrayOfNumbers[i] = +prompt(
+      `Ви вели не вірний тип даних! Спобуйте знову додати число до масиву під індексом ${i}`
+    );
+  } else {
+    newArrayOfNumbers[i] = userNumber;
+  }
+  if (isNaN(newArrayOfNumbers[i])) {
+    i--;
+    console.log(
+      "Ви вели не число, сробуйте знову з початку! Тільки будьласка уважніше!!!)))"
+    );
+  }
+}
+
+newArrayOfNumbers.sort(function (a, b) {
+  return a - b;
+});
+
+const shoppingList = [
+  { productName: "bread", productPrice: 25, productQuantity: 2 },
+  { productName: "beer", productPrice: 30, productQuantity: 4 },
+  { productName: "potatoes", productPrice: 8.5, productQuantity: 4 },
+  { productName: "tea", productPrice: 80, productQuantity: 2 },
+  { productName: "coffee", productPrice: 16, productQuantity: 2 },
+  { productName: "cheese", productPrice: 75, productQuantity: 1 },
+  { productName: "bananas", productPrice: 16, productQuantity: 4 },
+];
+
+let totalCostOfPurchases = 0;
+shoppingList.forEach((element) => {
+  totalCostOfPurchases += element.productQuantity * element.productPrice;
+});
+console.log(`Загальна сума покупок ${totalCostOfPurchases} грн.`); // 535
+
+const productQuantityArray = shoppingList.map((element) => {
+  return element.productQuantity;
+});
+const minNumberOfProducts = Math.min(...productQuantityArray); // 1
+
+let totalNumberOfProducts = 0;
+shoppingList.forEach((element) => {
+  totalNumberOfProducts += element.productQuantity;
+}); // 19
+
+let maxPriceOfProduct = 0;
+shoppingList.forEach((element) => {
+  if (element.productPrice > maxPriceOfProduct) {
+    maxPriceOfProduct = element.productPrice;
+  }
+});
+// maxPriceOfProduct = 80
+let productWithMaxPrice;
+shoppingList.forEach((element) => {
+  if (element.productPrice === maxPriceOfProduct) {
+    productWithMaxPrice = element.productName;
+  }
+});
+// productWithMaxPrice = tea
+
+const addingProductToList = () => {
+  const newProduct = {
+    productName: prompt(
+      "Ведіть назву продукту який ви хочете додати до списку"
+    ).toLowerCase(),
+    productPrice: +prompt("Ведіть ціну продукту"),
+    productQuantity: +prompt("Ведіть кількість продукту"),
```

Comment:
Nice way of doing so, but here much safer is to split

`const objectFields = ['productName', 'productPrice', productQuantity']`

Then run .forEach on that array and populate the object with `prompt` (but `prompt` separately)

Try to use feature called - property shorthand

```suggestion
    productName,
    productPrice,
    productQuantity,
```